pkg/sql/show_syntax.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package sql
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/lex"
+	"github.com/cockroachdb/cockroach/pkg/sql/parser"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/pkg/errors"
+)
+
+// ShowSyntax implements the plan for SHOW SYNTAX. This statement is
+// usually handled as a special case in Executor, but for
+// FROM [SHOW SYNTAX ...] we will arrive here too.
+func (p *planner) ShowSyntax(ctx context.Context, n *tree.ShowSyntax) (planNode, error) {
+	var query bytes.Buffer
```

Comment:
I think it would help maintainability to put a full example of the query you're constructing in a comment. Otherwise it's hard to figure out the overall flow of what's going on without reading all of the fmt.Fprintf and conditionals below.